pkg/test/framework/components/istio/ingress/interface.go

```diff
@@ -81,9 +81,9 @@ type Instance interface {
 	// TCPAddress returns the external TCP (31400) address of the ingress gateway (or the NodePort address,
 	// when in an environment that doesn't support LoadBalancer).
 	TCPAddress() net.TCPAddr
-	// DiscoveryAddress returns the external XDS (!5012) address on the ingress gateway (or the NodePort address,
+	// DiscoveryAddress returns the external XDS (!5012) address for the given service (or the NodePort address,
 	// when in an evnironment that doesn't support LoadBalancer).
-	DiscoveryAddress() net.TCPAddr
+	DiscoveryAddress(svcName string) net.TCPAddr
```

Comment:
I don't think changing this API is the right thing to do. If we have the `configTopology` as I described above, then we should just do the right thing based on the topology.